paper.md

```diff
@@ -2,11 +2,77 @@
+Scientific workflow engines have been a predominant solution [@stevens2013automated] [@banati2015minimal] [@qasha2016framework] for handling the reproducibility problem by organizing the steps in a complex scientific workflow as the nodes of a directed acyclic graph (DAG) and executing them in correct order [@albrecht2012makeflow].
+Nextflow [@ditommaso_nextflow_2017], Pegasus [@deelman_pegasus_2004] and Taverna [@oinn_taverna_2004] are some popular examples of scientific workflow engines.
+But some phenomena like unavailability of third-party services, missing example input data, changes in the execution environment, insufficient documentation of workflows make it difficult for scientists to reuse workflows, thus causing what is known as _workflow decay_ [@workflow_decay].
+
+<!-- attempts to solves using containers  -->
+
+One of the main reasons behind _workflow decay_ is the differences in the environment where the workflows are developed and where they are reproduced [@meng2017facilitating]. 
+VM's were used to address this problem for some time due to their high isolation gurantees, where every task ran inside a separate VM [@howe2012virtual] [@virtandnestedvirt2012]. 
+Since VM's had large resource footprints, researchers replaced VM's with Software containers i.e. with light-weight virtualization technologies to provide platform-independent reproducibility [@barik2016performance] [@sharma2016containers].
+Although Software (Linux) containers are a relatively old technology [@menage_adding_2007], it was not until recently, with the rise of Docker, that they entered mainstream territory [@bernstein_containers_2014]. 
+Since then, this technology has transformed the way applications get deployed in shared infrastructures, with 25% of companies using this form of software deployment [@datadog_surprising_2018], and a market size projected to be close to 5B by 2023 [@marketsandmarkets_application_2018]. 
+Docker has been the de facto container runtime, with other container runtimes such as Singularity [@kurtzer_singularity_2017], Rkt [@rktcommunity_rkt_2019], Charliecloud [@priedhorsky_charliecloud_2017] and Podman [@podmancommunity_containers_2019] having emerged. 
+The Linux Foundation bootstrapped the Open Container Initiative (OCI) [@opencontainerinitiative_new_2016] and is close to releasing version 1.0 of a container image and runtime specifications. 
+With Docker, began the era of container-native software development, which is a paradigm that promotes the building, testing, and deployment of software in containers. 
```

Comment:
"With Docker, the container-native software development paradigm emerged, which ..."